src/pkg/robot2/dao/dao.go

```diff
@@ -0,0 +1,144 @@
+package dao
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/orm"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/robot2/model"
+	"time"
+)
+
+// DAO defines the interface to access the robot data model
+type DAO interface {
+	// Create ...
+	Create(ctx context.Context, r *model.Robot) (int64, error)
+
+	// Update ...
+	Update(ctx context.Context, r *model.Robot, props ...string) error
+
+	// Get ...
+	Get(ctx context.Context, id int64) (*model.Robot, error)
+
+	// Count returns the total count of robots according to the query
+	Count(ctx context.Context, query *q.Query) (total int64, err error)
+
+	// List ...
+	List(ctx context.Context, query *q.Query) ([]*model.Robot, error)
+
+	// Delete ...
+	Delete(ctx context.Context, id int64) error
+
+	// DeleteByProjectID ...
+	DeleteByProjectID(ctx context.Context, projectID int64) error
+}
+
+// New creates a default implementation for Dao
+func New() DAO {
+	return &dao{}
+}
+
+type dao struct{}
+
+func (d *dao) Create(ctx context.Context, r *model.Robot) (int64, error) {
+	ormer, err := orm.FromContext(ctx)
+	if err != nil {
+		return 0, err
+	}
+	r.CreationTime = time.Now()
+	id, err := ormer.Insert(r)
+	if err != nil {
+		return 0, orm.WrapConflictError(err, "robot account %d:%s already exists", r.ProjectID, r.Name)
+	}
+	return id, err
+}
+
+func (d *dao) Update(ctx context.Context, r *model.Robot, props ...string) error {
+	ormer, err := orm.FromContext(ctx)
+	if err != nil {
+		return err
+	}
+	r.UpdateTime = time.Now()
+	n, err := ormer.Update(r, props...)
```

Comment:
`UpdateTime` needs to append to `props` when `props` is not empty otherwise `UpdateTime` will not be updated when props provided.

Reply:
remove it, and let the caller to set it